src/yoast-model.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Yoast extension of the Model class.
+ *
+ * @package Yoast\YoastSEO
+ */
+
+namespace Yoast\YoastSEO;
+
+use YoastSEO_Vendor\Model;
+use YoastSEO_Vendor\ORMWrapper;
+
+/**
+ * Make Model compatible with WordPress.
+ */
+class Yoast_Model extends Model {
+	/**
+	 * The table name for the implemented Model.
+	 *
+	 * @var string $_table
+	 */
+	public static $_table;
+
+	/**
+	 * Hacks around the Model to provide WordPress prefix to tables.
+	 *
+	 * @param string $class_name   Type of Model to load.
+	 * @param bool   $yoast_prefix Optional. True to prefix the table name with the Yoast prefix.
+	 *
+	 * @return ORMWrapper Wrapper to use.
+	 */
+	public static function of_type( $class_name, $yoast_prefix = true ) {
+		// Prepend namespace to the class name.
+		$class = self::$auto_prefix_models . $class_name;
+
+		// Set the class variable to the custom value based on the WPDB prefix.
+		$class::$_table = self::get_table_name( $class_name, $yoast_prefix );
```

Comment:
Minor multisite issue just to be aware of: Since the table name is stored at this point, we need to ensure that the `ORMWrapper` returned by the method is not used across possible site switching processes.

This example is not problematic and will work:
```php
switch_to_blog( 3 );

$indexable = Yoast_Model::of_type( 'Indexable' )
	->where( 'object_type', 'post' )
	->find_one();
```

This example would return an unexpected result (it would still look at the database table from the original site instead of from the site with ID 3):
```php
$orm = Yoast_Model::of_type( 'Indexable' );

switch_to_blog( 3 );

$indexable = $orm
	->where( 'object_type', 'post' )
	->find_one();
```

I think it's fine as it currently works, but we should keep that behavior in mind.